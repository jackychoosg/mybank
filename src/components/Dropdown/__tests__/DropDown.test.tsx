import { render, cleanup, fireEvent } from '@testing-library/react'
import { DropDown } from '../index'
describe('DropDown component', () => {
  afterEach(cleanup)
  const items = [
    {
      id: 1,
      name: 'account1'
    },
    {
      id: 2,
      name: 'account2'
    }
  ]
  const onDropdownChange = jest.fn()
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <DropDown
        label="Account"
        items={items}
        selectedItem={0}
        onChange={onDropdownChange}
        defaultOptionLabel="--Choose the item--"
        errorMessage="Required"
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onChange action', () => {
    const { getByTestId, getAllByTestId } = render(
      <DropDown
        label="Account"
        items={items}
        selectedItem={1}
        onChange={onDropdownChange}
        defaultOptionLabel="--Choose the item--"
      />
    )
    let options = getAllByTestId('select-option') as HTMLOptionElement[]
    expect(options[0].selected).toBeFalsy()
    expect(options[1].selected).toBeTruthy()
    expect(options[2].selected).toBeFalsy()
    fireEvent.change(getByTestId('select'), { target: { value: 2 } })
    expect(onDropdownChange).toBeCalled()
  })
})
