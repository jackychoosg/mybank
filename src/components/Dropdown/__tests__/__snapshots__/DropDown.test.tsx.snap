// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT ckIJnv"
  >
    <label
      class="sc-dlnjwi cVzCZm"
    >
      Account
      <div
        class="sc-bdnxRM bqGtyK"
      >
        Required
      </div>
    </label>
    <select
      class="sc-hKFxyN bVnZyW"
      data-testid="select"
    >
      <option
        class="sc-eCApnc"
        data-testid="select-option"
        value="0"
      >
        --Choose the item--
      </option>
      <option
        class="sc-eCApnc"
        data-testid="select-option"
        value="1"
      >
        account1
      </option>
      <option
        class="sc-eCApnc"
        data-testid="select-option"
        value="2"
      >
        account2
      </option>
    </select>
  </div>
</DocumentFragment>
`;
